fiftyone/utils/eval/segmentation.py

```diff
@@ -495,21 +513,34 @@ def _from_dict(cls, d, samples, config, eval_key, **kwargs):
         )
 
     @staticmethod
-    def _parse_confusion_matrix(confusion_matrix, classes):
+    def _parse_confusion_matrix(confusion_matrix, classes, ytrue_ids_dict, ypred_ids_dict):
         ytrue = []
         ypred = []
         weights = []
+        ytrue_ids = None
+        ypred_ids = None
+        if ytrue_ids_dict is not None and ypred_ids_dict is not None:
+            ytrue_ids = []
+            ypred_ids = []
         nrows, ncols = confusion_matrix.shape
         for i in range(nrows):
             for j in range(ncols):
+                index = (i, j)
                 cij = confusion_matrix[i, j]
                 if cij > 0:
-                    ytrue.append(classes[i])
-                    ypred.append(classes[j])
-                    weights.append(cij)
-
-        return ytrue, ypred, weights
-
+                    if ytrue_ids_dict is not None and ypred_ids_dict is not None:
+                        ytrue_ids+= ytrue_ids_dict[index]
+                        ypred_ids+= ypred_ids_dict[index]
+                        ytrue_multiplier = len(ytrue_ids_dict[index])
+                    else:
+                        ytrue_multiplier = 1
+                    for p in range(ytrue_multiplier):
+                        ytrue.append(classes[i])
+                        ypred.append(classes[j])
+                        weights.append(cij/ytrue_multiplier)
+                    #Note: The weights aren't equally divided by the different ytrue_ids, but it works out for the confusion matrix calculations.
```

Comment:
Agree, ideally the weights would be the actual weights/counts from each pair of (ytrue_ids, ypred_ids). It seems to work okay as-is but long-run there may be other ramifications. For instance, one idea, if I click on a misclassification (off-diagonal) cell, ideally I would see the resulting samples sorted in descending order of weight, with the first sample containing the most misclassified pixels of that type.

Reply:
Fixed and updated in latests commit